updateHostsFile.py

```diff
@@ -325,6 +326,34 @@ def main():
             flush_cache=settings["flushdnscache"], prompt_flush=not auto
         )
 
+def await_network_connection(retries=3, delay=10):
+    """
+    Check if we have a proper name resolution and connectivity
+    By pinging a common domain
+
+    Parameters
+    ----------
+    retries : int
+        Number of times to retry again (on ping failure)
+    delay : int
+        How many seconds to wait before retrying again (on ping failure)
+
+    Returns
+    -------
+    -> bool
+        True : A ping succeded
+        False : All retries have failed
+    """
+    while retries:
+        if 0 == subprocess.call(['ping', 'one.one.one.one', '-c', '1'],
```

Comment:
The script should run on windows too.

Please assume that on windows `ping` will expect `-n` instead of `-c`.

So.
```python
import platform

if platform.system().lower() == "windows":
     count_arg = "-n"
else:
     count_arg = "-c"
```